### ISSUE TITLE
Errors throwed on minified (onelined) CSS are unreadable

### ISSUE THREAD

**Issue by Den-dp** [postcss/postcss#1962]
Current pretty printing for errors works poorly for long lines on minified CSS.

![image](https://github.com/user-attachments/assets/48bc95eb-0665-4ebd-8aba-2237715472eb)

> I found it when I started to write `postcss` plugins that apply different optimizations and rewrite minified CSS files provided by 3rd parties.

Since I tried to stick with the idea of failing fast if our plugins were unable to do needed stuff, I started to use `node.error` functionality as described in https://github.com/postcss/postcss/blob/main/docs/guidelines/plugin.md#41-use-nodeerror-on-css-relevant-errors

Here is a simplified reproduction (but in reality, I use `webpack`):

```js
import postcss from "postcss";

function postcssFontFaceError() {
  return {
    postcssPlugin: "postcss-font-face-error",
    AtRule: {
      "font-face": (atRule) => {
        throw atRule.error("Usage of @font-face is not allowed");
      },
    },
  };
}

const longMinifiedCSS = `.foo {color: red;}`.repeat(100);
const fontFaceCSS = `@font-face{font-family:'Roboto';font-weight:600;font-style:normal;src:url('someurl') format('woff2')}`;

const css = fontFaceCSS.concat(longMinifiedCSS);

postcss({
  plugins: [postcssFontFaceError()],
})
  .process(css, { from: "./styles.css" })
  .then((result) => {
    console.log(result.css);
  })
  .catch((error) => {
    console.error(error.toString());
  });
```

**Comment by Den-dp** [comment]
@ai so it's not about the colors; it's about the length.

Since, in addition to the node with the error (`@font-face{...}`), the entire line of a file is also displayed, the larger the minified (single-lined) CSS file, the less readable the error becomes.

In my use case, a processed file can reach hundreds of kilobytes, which leads to very big error spread to multiple screens.

So while pretty-printing, it would be helpful to display only the node with error or at least the small portion of the line

**Comment by ai** [comment]
Got it. In this case we can show only -20…+20 symbols around the error if the line is too big.

Can I ask you to send PR?

**Comment by ai** [comment]
If you do not have time, I can ask on Cult of Martians

**Comment by Den-dp** [comment]
After yesterday's massive attack on energetic infrastructure I have very limited access to electricity and the internet, so my initial intent to contribute it this weekend looks unrealistic.

So if you have a capacity for it, that would be helpful!

**Comment by ai** [comment]
Oh, fuck. I am very sorry. 

I am going to fix it in Cult of Martians. It will be fixed in a week or two.

**Comment by varpstar** [comment]
Opened PR with fix:  https://github.com/postcss/postcss/pull/1965

**Comment by ai** [comment]
Fixed in 8.4.42

**Comment by Den-dp** [comment]
I just checked, and it works great! Thank you all
